GUI/src/main/java/com/example/application/views/main/LoginView.java

```diff
@@ -24,18 +25,28 @@ public class LoginView extends LoginOverlay {
     private SubscribedUserService subscribedUserService;
     public LoginView() {
         service = (UserService)Load("service");
+        subscribedUserService = new SubscribedUserServiceImp(null);
         setTitle(projectName);
         setDescription(projectDescription);
         setOpened(true);
         addLoginListener(e -> {
-            var res = service.login(e.getUsername(), e.getPassword());
+            var res = subscribedUserService.login(e.getUsername(), e.getPassword());
             if (res.isOk()) {
-                subscribedUserService = res.getElement();
-                save("service",subscribedUserService);
-                save("user-name", e.getUsername());
-                save("subscribed-user-service", subscribedUserService);
                 notifySuccess("Login Succeeded!");
-                UI.getCurrent().navigate(SubscribedUserView.class);
+                subscribedUserService = res.getElement();
+                var resManager = subscribedUserService.manageSystemAsSystemManager();
+                if(resManager.isOk()){
+                    SystemManagerService managerService = resManager.getElement();
+                    save("system-manager-service", managerService);
```

Comment:
still there need to be only one service at all time in save("service")

Reply:
neew to be : 
save("user-name", e.getUsername());
if(resManager.isOk()){
  save("service", resManager.getElement());
  UI.getCurrent().navigate(SystemManagerView.class);
}
else {
  save("service", subscribedUserService);
  UI.getCurrent().navigate(SubscribedUserView.class);
}